### DISCUSSION TITLE
CapsuleTriangleTester wrong result when the capsule internal line segment touching the edge

### DISCUSSION THREAD

**Discussion by lmtss** [bepu/bepuphysics2#250] [General]
I'm learning about narrowphase, and I found that CapsuleTriangleTester doesn't perform well in a scene.
![image](https://user-images.githubusercontent.com/22532316/208618858-1fe49076-7891-4615-894e-6eeaf90591b7.png)
![image](https://user-images.githubusercontent.com/22532316/208618892-92379268-3b74-462a-b16c-a85aeea6685e.png)
![image](https://user-images.githubusercontent.com/22532316/208618987-c230befe-125e-4dbf-b953-95ec25495658.png)
The edge of the triangle intersects the axis of the capsule. According to the code, fallback should be used at this time. But the square of the shortest distance between two line segments is 8e-15, which is greater than the threshold (1e-15), so fallback will not be performed.
```c#
// normalLengthSquared == 8e-15
var useFallbackNormal = Vector.LessThan(normalLengthSquared, new Vector<float>(1e-15f));
```  
I think in this case, a fallback should be performed, but I don't know how to solve this problem, maybe simply modify the threshold?

The scene:
```c#
var capsule = new Capsule(0.5f, 2.0f);
var inertia = capsule.ComputeInertia(1);
var shapeIndex = Simulation.Shapes.Add(capsule);

Simulation.Bodies.Add(BodyDescription.CreateDynamic(new Vector3(0, 0, 0), new Vector3(0, 0, 0), inertia, shapeIndex, 0.01f));

var triangles = new QuickList<Triangle>(1, BufferPool);
var v0 = new Vector3(-1.0f, 0, -1.0f);
var v1 = new Vector3(-1.0f, 0, 1.0f);
var v2 = new Vector3(1.0f, 0, -1.0f);
triangles.AllocateUnsafely() = new Triangle { A = v0, B = v2, C = v1 };
var testMesh = new Mesh(triangles, Vector3.One, BufferPool);
Simulation.Statics.Add(new StaticDescription(new Vector3(0, 0, 0), Simulation.Shapes.Add(testMesh)));
```
2.4.0

**Comment by RossNordby** [comment]
A slightly more forgiving threshold should be fine, though on top of that, there's a lil bug with the first fallback. It requires calibration, and that calibration was removed in an earlier version, so it ended up pointing the wrong direction depending on the capsule's orientation. Since that normal is generally very poor, it ended up choosing another one (like the triangle normal).

Should be fixed in 4d87f41f56dbc321c8cde4219ff2889b40dde7f9, thanks for the report!
